driver-core/src/main/com/mongodb/MongoClientSettings.java

```diff
@@ -613,9 +599,15 @@ private MongoClientSettings(final Builder builder) {
         clusterSettings = builder.clusterSettingsBuilder.build();
         serverSettings = builder.serverSettingsBuilder.build();
         socketSettings = builder.socketSettingsBuilder.build();
-        heartbeatSocketSettings = builder.heartbeatSocketSettingsBuilder.build();
         connectionPoolSettings = builder.connectionPoolSettingsBuilder.build();
         sslSettings = builder.sslSettingsBuilder.build();
         compressorList = builder.compressorList;
+        SocketSettings.Builder heartbeatSocketSettingsBuilder = SocketSettings.builder()
+                .readTimeout(20000, MILLISECONDS)
```

Comment:
I believe SDAM requires that the readTimeout for monitor connections is set to the same value as the connect timeout, rather than hardcoded to 20 seconds

Reply:
https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#socket-timeout-for-monitoring-is-connecttimeoutms